### PR TITLE
Fix/challenge based on weight

### PR DESCRIPTION
**Pull request by Jayashsatolia403**

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | sprint-1.11  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  |   |
| zwalletcli  |   |
| 0chain  |   |
| blobber  |   |
| authorizer  |   |
| 0box  | fix/graph-endpoints  |
| 0dns  |   |

File: .github/workflows/nightly-challenge-sprint.yaml [hunk 1]
```diff
@@ -52,38 +53,3 @@ jobs:
           S3_ACCESS_KEY: ${{ secrets.S3_ACCESS_KEY }}
           S3_SECRET_KEY: ${{ secrets.S3_SECRET_KEY }}
 
-  notify_slack_on_failure:
-    runs-on: [self-hosted, arc-runner]
-    needs: [system-tests]
-    if: always() && (needs.system-tests.result == 'failure')
-    steps:
-      - name: "Notify Slack"
```
Comment (stewartie4) [inline .github/workflows/nightly-challenge-sprint.yaml:60]: Why is this being removed

File: .github/workflows/nightly-challenge-sprint.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
           echo "RUNNER_NUMBER=${RUNNER_NAME:(-1)}" >> $GITHUB_ENV
 
       - name: "Deploy 0Chain"
-        uses: 0chain/actions/deploy-0chain@master
```
Comment (stewartie4) [inline .github/workflows/nightly-challenge-sprint.yaml:23]: Revert

Comment (stewartie4) [inline .github/workflows/nightly-challenge-sprint.yaml:23]: Revert

**Review by stewartie4 — CHANGES_REQUESTED** [review]

File: .github/workflows/nightly-challenge-sprint.yaml [hunk 3]
```diff
@@ -35,6 +35,7 @@ jobs:
       - name: "Run Challenge System tests"
         uses: 0chain/actions/run-system-tests@master
         with:
+          system_tests_branch: ${{ env.CURRENT_BRANCH }}
```
Comment (stewartie4) [inline .github/workflows/nightly-challenge-sprint.yaml:38]: What is this change for?

Comment (Jayashsatolia403) [reply .github/workflows/nightly-challenge-sprint.yaml:38]: So we can use run it manually with current branch. It should work right?

Comment (Jayashsatolia403) [reply .github/workflows/nightly-challenge-sprint.yaml:38]: Can it break cron job runs?

Comment (Kishan-Dhakan) [reply .github/workflows/nightly-challenge-sprint.yaml:38]: CURRENT_BRANCH is not exported anywhere

Comment (stewartie4) [reply .github/workflows/nightly-challenge-sprint.yaml:38]: There is no current branch variable set - the CI should default to the current sprint if nothing is supplied which is what was happening before

Comment (Jayashsatolia403) [reply .github/workflows/nightly-challenge-sprint.yaml:39]: My bad. Pushed a fix. Really sorry for inconvenience.

Comment (stewartie4) [reply .github/workflows/nightly-challenge-sprint.yaml:39]: There is no current branch variable set - the CI should default to the current sprint if nothing is supplied which is what was happening before

**Review by Kishan-Dhakan — APPROVED** [review]

**Pull request merged**